hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/timeline/HoodieTimelineArchiver.java

```diff
@@ -111,13 +114,13 @@ public boolean archiveIfRequired(HoodieEngineContext context, boolean acquireLoc
         };
         this.timelineWriter.write(instantsToArchive, Option.of(action -> deleteAnyLeftOverMarkers(context, action)), Option.of(exceptionHandler));
         LOG.info("Deleting archived instants " + instantsToArchive);
-        success = deleteArchivedInstants(instantsToArchive, context);
+        deleteArchivedInstants(instantsToArchive, context);
         // triggers compaction and cleaning only after archiving action
         this.timelineWriter.compactAndClean(context);
       } else {
         LOG.info("No Instants to archive");
       }
-      return success;
+      return instantsToArchive.size();
```

Comment:
Should we also include the `success` flag also as a metric?

Reply:
In the current archive code, it seems that `success` is always set to true. The `success` variable is initialized as true, and `deleteArchivedInstants` always returns true unless it fails. However, if there is a failure, the current implementation should throw an exception and terminate without returning any value. Therefore, I believe the `success` variable is meaningless in the current context. Alternatively, should we catch these exceptions and return false? I'm not sure if this would be reasonable.

Reply:
We can address it in another PR, we kind of do not have atomicity for current 2 steps:

- flush of archived timeline
- deletion of active metadata files

My roughly thought about it, we can fix the left over active metadata files (should be deleted from active timeline) in the next round of archiving, imagine the latest instant time in archived timeline is t10 and the oldest instant in active timeline is t7, we should retry the deletion of instant metadat files from t7 ~ 10 at the very beginning.